src/main/java/com/algorand/algosdk/v2/client/indexer/LookupAssetTransactions.java

```diff
@@ -0,0 +1,286 @@
+package com.algorand.algosdk.v2.client.indexer;
+
+import java.io.IOException;
+
+import com.algorand.algosdk.v2.client.connect.Client;
+import com.algorand.algosdk.v2.client.connect.Query;
+import com.algorand.algosdk.v2.client.connect.QueryData;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.algorand.algosdk.v2.client.model.TransactionsResponse;
+
+
+/**
+ * Lookup transactions for an asset. /assets/{asset-id}/transactions 
+ */
+public class LookupAssetTransactions extends Query {
+	private String address;
+	private String addressRole;
+	private String afterTime;
+	private long assetId;
+	private String beforeTime;
+	private long currencyGreaterThan;
+	private long currencyLessThan;
+	private boolean excludeCloseTo;
+	private long limit;
+	private long maxRound;
+	private long minRound;
+	private String next;
+	private String notePrefix;
+	private long round;
+	private String sigType;
+	private String txId;
+	private String txType;
+
+	private boolean addressIsSet;
+	private boolean addressRoleIsSet;
+	private boolean afterTimeIsSet;
+	private boolean assetIdIsSet;
+	private boolean beforeTimeIsSet;
+	private boolean currencyGreaterThanIsSet;
+	private boolean currencyLessThanIsSet;
+	private boolean excludeCloseToIsSet;
+	private boolean limitIsSet;
+	private boolean maxRoundIsSet;
+	private boolean minRoundIsSet;
+	private boolean nextIsSet;
+	private boolean notePrefixIsSet;
+	private boolean roundIsSet;
+	private boolean sigTypeIsSet;
+	private boolean txIdIsSet;
+	private boolean txTypeIsSet;
+
+	public LookupAssetTransactions(Client client) {
+		super(client);
+	}
+
+	/**
+	 * Only include transactions with this address in one of the transaction fields. 
+	 */
+	public LookupAssetTransactions setAddress(String address) {
+		this.address = address;
+		this.addressIsSet = true;
+		return this;
+	}
+
+	/**
+	 * Combine with the address parameter to define what type of address to search for. 
+	 */
+	public LookupAssetTransactions setAddressRole(String addressRole) {
+		this.addressRole = addressRole;
+		this.addressRoleIsSet = true;
+		return this;
+	}
+
+	/**
+	 * Include results after the given time. Must be an RFC 3339 formatted string. 
+	 */
+	public LookupAssetTransactions setAfterTime(String afterTime) {
+		this.afterTime = afterTime;
+		this.afterTimeIsSet = true;
+		return this;
+	}
+	public LookupAssetTransactions setAssetId(long assetId) {
+		this.assetId = assetId;
+		this.assetIdIsSet = true;
+		return this;
+	}
+
+	/**
+	 * Include results before the given time. Must be an RFC 3339 formatted string. 
+	 */
+	public LookupAssetTransactions setBeforeTime(String beforeTime) {
```

Comment:
Using a string is not very friendly, this should be something like Date, DateTime, or a Joda type

Reply:
It will be helpful to have this specified in the spec so that all SDKs can implement a similar interpretation of the field. 
This can also be done for other fields, e.g. base 64 encodings, dates, etc.

Reply:
This particular example is outside of the spec. The server accepts a string with a specific format, the SDK can provide convenience methods to set that string in different ways.

Reply:
The spec does provide a "format" type, we can come up with some sort of format convention if it would make this sort of thing easier, I really like the idea of automatically generating different setters based on the spec

Reply:
This is also true for things like addresses, we should be able to accept an address object.

Reply:
I think we'll need to use a custom convention, it looks like OpenAPI supports adding arbitrary fields like `x-algorand-format: RFC3339 String`.

Reply:
@algonautshant added `x-algorand-format` to some of the fields.

I also added a `uint64` type for BigInteger detection.

I'm sure I missed a couple, let me know if you find any.

https://github.com/algorand/openapi-server-generator/blob/2517f43cda3e2d1d7f29c12a790fd5972df62287/scripts/indexer.oas2.yml#L653